WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java

```diff
@@ -35,8 +36,44 @@
      * -- 12.9 OPTED OUT (were auto-opted in but turned it OFF) -> turn all sites OFF in 13.0
      *
      */
-    public static void migrateAppWideMobileEditorPreferenceToRemote(final Context context,
+    public static void migrateAppWideMobileEditorPreferenceToRemote(final AccountStore accountStore,
+                                                                    final SiteStore siteStore,
                                                                     final Dispatcher dispatcher) {
+        // In a later version we might override mobile_editor setting if it's set to `aztec` and show a specific notice
+        // for these users ("We made a lot of progress on the block editor and we think it's now better than
+        // the classic editor, we switched it on, but you can change the configuration in your Site Settings").
+        // ^ This code should be here.
+
+        // If the user is already in the rollout group, we can skip this the migration.
+        if (AppPrefs.isUserInGutenbergRolloutGroup()) {
+            return;
+        }
+
+        // Check if the user has been "randomly" selected to enter the rollout group.
+        //
+        // For self hosted sites, there are often one or two users, and the user id is probably 0, 1 in these cases.
+        // If we exclude low ids, we won't get an not an homogeneous distribution over self hosted and WordPress.com
+        // users, but the purpose of this is to do a progressive rollout, not an necessarily an homogeneous rollout.
+        //
+        // To exclude ids 0 and 1, to rollout for 10% users,
+        // we'll use a test like `id % 100 >= 90` instead of `id % 100 < 10`.
+        if (accountStore.getAccount().getUserId() % 100 >= (100 - GB_ROLLOUT_PERCENTAGE)) {
+            // We want to make sure to enable Gutenberg only on the sites they didn't opt-out.
+            for (SiteModel site : siteStore.getSites()) {
+                if (TextUtils.isEmpty(site.getMobileEditor())) {
+                    // Enable Gutenberg
+                    enableBlockEditor(dispatcher, site);
```

Comment:
The code in `enableBlockEditor` does make a network call to the backend. If you have many sites in the app there is a spike in network calls, and some of those can even fail...(We've experienced this problem when migrated the Editor pref from app-wide to site settings).
We should re-use the action `SiteActionBuilder.newDesignateMobileEditorForAllSitesAction`, like here: https://github.com/wordpress-mobile/WordPress-Android/pull/10752/files#diff-cb360ad2878e66ffe65d8c54967921d6R83

Reply:
Will this action override all sites configuration? Even sites that have been switched back to Aztec? (that's the reason why I used `enableBlockEditor` instead).

Reply:
Yes, it does override all sites configuration.  I think there is a way to by-pass this behavior, we need to add a new action to FluxC that does set the parameter `set_only_if_empty` to `true` on the underlying network call. 
Unfortunately making sequential network calls, even if you delay it from the client, may run into race condition on the PHP backend, and there is no guarantee that the call succeed as expected.

Make sure to test via Developer console first, to check if the parameter is working fine or not.

Reply:
I'll try that, and there is actually another thing we could do: check if there is one `aztec` setting for any of the user's sites:
- If there is none, then enable gutenberg for all sites. 
- If there is at least one, then don't put this user in the rollout cohort.

Reply:
I like your idea above, it should help on making the logic simpler, and dones't required FluxC changes.

We should also make sure the device has network connectivity, since the action is "fired" but we're not listening for errors I guess. So better to try when the device is online only.

Reply:
Logic updated in 2ff8e8d - users who have at least one aztec enabled site will be excluded from the cohort